workshops/check.py

```diff
@@ -324,82 +377,57 @@ def check_categories(left, right, errors, error_msg):
     return True
 
 
-def check_repeated_categories(seen_categories, errors, error_msg):
-    '''Check for categories appearing two or more times.'''
-    category_counts = Counter(seen_categories)
-    double_categories = [category for category in category_counts
-                         if category_counts[category] > 1]
+def get_header(text):
+    '''Extract YAML header from raw data, returning (None, None) if no
+    valid header found and (raw, parsed) if header found.'''
 
-    if double_categories:
-        add_error(error_msg, errors)
-        msg = '"{0}" appears more than once'.format(double_categories)
-        add_suberror(msg, errors)
-        return False
-
-    return True
+    # YAML header must be right at the start of the file.
+    if not text.startswith('---'):
+        return None, None
 
+    # YAML header must start and end with '---'
+    pieces = text.split('---')
+    if len(pieces) < 2:
+        return None, None
```

Comment:
raw = pieces[1].strip()
return raw, yaml.load(raw)

means that "key: value\n   " will be converted to "key:value", which means that those leading spaces won't be seen by yaml.load.